AFQ/segmentation.py

```diff
@@ -69,22 +69,87 @@ def segment(fdata, fbval, fbvec, streamlines, bundles=AFQ_BUNDLES,
         mapping = reg.read_mapping(mapping, img, reg_template)
 
     fiber_groups = {}
-    for hemi in ["R", "L"]:
-        for bundle in bundles:
-            ROIs = [bundle + "_roi%s_" % (i + 1) + hemi for i in range(2)]
-            select_sl = xform_sl
-            for ROI in ROIs:
-                data = templates[ROI].get_data()
-                warped_ROI = patch_up_roi(mapping.transform_inverse(
-                    data,
-                    interpolation='nearest'))
-
-                select_sl = dts.select_by_rois(select_sl,
-                                               [warped_ROI.astype(bool)],
-                                               [True])
-            if as_generator:
-                fiber_groups[bundle + "_" + hemi] = select_sl
-            else:
-                fiber_groups[bundle + "_" + hemi] = list(select_sl)
+    for bundle in bundles:
+        select_sl = xform_sl
+        for ROI, rule in zip(bundles[bundle]['ROIs'], bundles[bundle]['rules']):
+            data = ROI.get_data()
+            warped_ROI = patch_up_roi(mapping.transform_inverse(
+                data,
+                interpolation='nearest'))
+            # This function requires lists as inputs:
+            select_sl = dts.select_by_rois(select_sl,
+                               [warped_ROI.astype(bool)],
+                               [rule])
+        # Next, we reorient each streamline according to an ARBITRARY, but
+        # CONSISTENT order. To do this, we use the first ROI for which the rule
+        # is True as the first one to pass through, and the last ROI for which
+        # the rule is True as the last one to pass through:
+
+        # Indices where the 'rule' is True:
+        idx = np.where(bundles[bundle]['rules'])
+
+        orient_ROIs = [bundles[bundle]['ROIs'][idx[0][0]],
+                       bundles[bundle]['ROIs'][idx[0][-1]]]
+
+        select_sl = dts.orient_by_rois(select_sl,
+                                       orient_ROIs[0].get_data(),
+                                       orient_ROIs[1].get_data(),
+                                       copy=False)
+        if as_generator:
+            fiber_groups[bundle] = select_sl
+        else:
+            fiber_groups[bundle] = list(select_sl)
 
     return fiber_groups
+
+
+def calculate_tract_profile(img, streamlines, affine=None, n_points=100,
+                            weighting=None):
+    """
+
+    Parameters
+    ----------
+    img : 3D volume
+
+    streamlines : list of arrays, or array
+
+    weighting : 1D array or 2D array (optional)
+
+    """
+    if isinstance(streamlines, list):
+        # Resample each streamline to the same number of points
+        # list => array
```

Comment:
perhaps `np.array`? I didn't get this at first... but maybe the issue is, I'm not in Python/Numpy enough these days :)